File: Day277/maxSumIncSubSeq.java
package Day277;

public class maxSumIncSubSeq {
    public static int maxSumInc(int n, int arr[]) {
        int answer = 0;
        int dp[] = new int[n];
        for (int i=0; i<n; i++) {
            dp[i] = arr[i];
        }
        for (int i=0; i<n; i++) {
            for (int j=0; j<i; j++) {
                if (arr[i] > arr[j]) {
                    dp[i] = Math.max(dp[i], dp[j]+arr[i]);
                }
            }
            answer = Math.max(answer, dp[i]);
        }
        return answer;
    }
    public static void main(String[] args) {
        int n1 = 5;
        int n2 = 4;
        int num1[] = {1, 101, 2, 3, 100};
        int num2[] = {4, 1, 2, 3};
        System.out.println(maxSumInc(n1, num1));
        System.out.println(maxSumInc(n2, num2));
    }
}

File: Day277/maxSumIncSubSeq.go
package main

import (
	"fmt"
	"math"
)

func maxSumInc (n int, num []int) int {
	a := make([]int, n)
	answer := 0 
	for  i,v := range num {
		a[i] = v
	}
	for i:=0; i<n; i++ {
		for j:=0; j<i; j++ {
			if num[i] > num[j] {
				a[i] = int(math.Max(float64(a[i]), float64(a[j] + num[i])))
			}
 		}
		answer = int(math.Max(float64(answer), float64(a[i])))
	}
	return answer
}

func main() {
	n1 := 5
	n2 := 4
	num1 := []int{1, 101, 2, 3, 100}
	num2 := []int{4, 1, 2, 3}
	fmt.Println(maxSumInc(n1, num1))
	fmt.Println(maxSumInc(n2, num2))
}